apps/src/templates/MultipleSectionsAssigner.jsx

```diff
@@ -0,0 +1,322 @@
+    updateHiddenScript(sectionId, scriptId, false);
+    assignToSection(
+      sectionId,
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId
+    );
+  };
+
+  // this is identical to unhideAndAssignUnit above but just has null as the scriptId
+  assignCourseWithoutUnit = section => {
+    const {
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      assignToSection
+    } = this.props;
+    const sectionId = section.id;
+    assignToSection(
+      sectionId,
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      null
+    );
+  };
+
+  isAssignableToSection = sectionParticipantType => {
+    return sectionParticipantType === this.props.participantAudience;
+  };
+
+  selectAllHandler = () => {
+    for (let i = 0; i < this.props.sections.length; i++) {
+      // if the section is NOT in currentSections assigned, assign it
+      const isSectionToBeAssigned = !this.state.currentSectionsAssigned.some(
+        s => s.code === this.props.sections[i].code
+      );
+      if (isSectionToBeAssigned) {
+        this.setState(state => {
+          const newList = [...state.currentSectionsAssigned];
+          newList.push(this.props.sections[i]);
+          return {currentSectionsAssigned: newList};
+        });
+      }
+    }
+  };
+
+  render() {
+    const {sections, assignmentName, onClose} = this.props;
+
+    return (
+      <BaseDialog isOpen={true} handleClose={onClose}>
+        <div style={styles.header} className="uitest-confirm-assignment-dialog">
+          {i18n.chooseSectionsPrompt({assignmentName})}
+        </div>
+        <div style={styles.content}>{i18n.chooseSectionsDirections()}</div>
+        <div style={styles.header} className="uitest-confirm-assignment-dialog">
+          {i18n.yourSectionsList()}
+        </div>
+        <hr />
+        <div style={styles.grid}>
+          {sections &&
+            sections.map(
+              section =>
+                this.isAssignableToSection(section.participantType) && (
+                  <TeacherSectionOption
+                    key={section.id}
+                    section={section}
+                    isChecked={
+                      !!this.state.currentSectionsAssigned.some(
+                        s => s.code === section.code
+                      )
+                    }
+                    assignedSections={this.state.currentSectionsAssigned}
+                    onChange={() => this.handleChangedCheckbox(section)} // this fucntion should update the state of multiple secion assigner
+                    editedValue={section.isAssigned}
+                  />
+                )
+            )}
+        </div>
+        <a
+          style={styles.selectAllSectionsLabel}
+          onClick={this.selectAllHandler}
+          className="select-all-sections"
+        >
+          Select All
+        </a>
+        <div style={{textAlign: 'right'}}>
+          <Button
+            __useDeprecatedTag
```

Comment:
I think you should be able to remove these `_useDeprecatedTag`s as well. Might be worth a quick scan of this PR to check for any leftover ones!